### DISCUSSION TITLE
SKR 1.4 turbo vs BTT TFT35 V2 (fw 2.0.27) vs RRF - No printer attached

### DISCUSSION THREAD

**Discussion by sssvolk** [bigtreetech/BIGTREETECH-TouchScreenFirmware#2764] [Q&A]
Hello everyone! 
There is no connection of the TFT display to the board. 
config.g - M575 P1 B57600 S1, parameter S changed from 0 to 4
board.txt - serial.aux.rxTxPins = { 0.3, 0.2 }
renamed from config_rrf.ini to config.ini for flashing of the firmware TFT

Why can't the display connect to the printer? What do I need to change in the settings?

**Comment by NewRay2022** [comment]
Yer it's a software but hardware problem, had my self with brand new 3d
printer with same specs, contact BIGTREETECH they should help you, find
right software or replace the hardware, if new otherwise I'm working on
printer got do test print and see if the software BIGTREETECH gave and a
replacement boad seams printer now shows, hopefully it helps.

**Comment by radek8** [comment]
You have to activate the serial port for SKR V1.4Turbo on the printer, it is port 0
You must set the same serial port communication speed on both the printer and the display.
Check if you have connected the TFT connector correctly and if the serial cable is not defective.

**Reply by sssvolk** [reply]
> You have to activate the serial port for SKR V1.4Turbo on the printer, it is port 0 You must set the same serial port communication speed on both the printer and the display. Check if you have connected the TFT connector correctly and if the serial cable is not defective.

The transfer rate is the same. The cable is intact. If I put the firmware 2.0.25 - the inscription "no printer" disappears, but there is no data from the printer. If I put the firmware 2.0.27 - the inscription "no printer" is lit.

**Reply by vin024github** [reply]
I have a similar problem when setting the printer to use klipper instead of Marlin… i upgrade the firmware to 1.27 and i still have the No printer attache problem but the Marlin display does not work either. All i see in Marlin mode is the title.

**Reply by vin024github** [reply]
How do you activate the serial port on the GD TFT35 B1 V3

**Reply by radek8** [reply]
You can activate the serial port on the TFT in two ways.
In the config.ini file, or from the display menu.

**Comment by vin024github** [comment]
I am using Klipper, there is no serial port definition and on the screen I dont find where to go to set the proper serial port. I found how to set the baudrate but not the serial port.Vincent BrouilletLe 13 déc. 2023 à 16:22, Radek ***@***.***> a écrit :﻿
You can activate the serial port on the TFT in two ways.
In the config.ini file, or from the display menu.

**Reply by radek8** [reply]
Yes, you can disable or enable the serial port on the display by defining the communication speed.
If you would like to change the serial ports for communication with the printer (for example, due to a non-functional pin on the default port - connector), it can be done with a simple modification and a new compilation of the FW.
Attention The FW of this display is optimized to work with Marlin. Clipper is not very supported.

**Comment by vin024github** [comment]
Radek,What do you mean by « not very supported «  when you talk about Klipper? It will not work at all? Bigtreetech is not going to do anything about it?Regards,Vincent BrouilletLe 14 déc. 2023 à 01:43, Radek ***@***.***> a écrit :﻿
Yes, you can disable or enable the serial port on the display by defining the communication speed.
If you would like to change the serial ports for communication with the printer (for example, due to a non-functional pin on the default port - connector), it can be done with a simple modification and a new compilation of the FW.
Attention The FW of this display is optimized to work with Marlin. Clipper is not very supported.

**Reply by radek8** [reply]
Sorry for my english, I'm using a translator.
BTT makes good HW but when it comes to tech support and SW it's bad.
SW for TFT is developed by a community of people (programmers) who like this display and want to improve it and publish modifications.
BBT mostly just merges changes, upgrades and fixes.
It hasn't even done that for the last year, that's why there are so many useful fixes waiting there.
Most programmers who work on the display use Marlin, so all changes are primarily compatible with Marlin.
Occasionally someone with a different system (Klipper, Smoothieware, RepRap) would sign up and want to add support for their system, but after two posts they wouldn't hear back and they hadn't finished the implementation.
The display will generally work with the Klipper. It will print and display temperature and such. but it may have a problem with some functions and settings (calibration, leveling, fan...).

Both Marlin and Klipper use the same G-codes, but some commands and some command responses have different syntax. If Klipper does not receive a command in the expected form, the display does not receive a response in the form it expects, this function will not work.
It could be solved, but it means modifying the program and compiling your own FW. But for that you need at least a basic knowledge of programming.

In the past, there was also a problem that Klipper did not display the native screen properly through the EXP connector, and BTT was working on making it operational, but I don't know if he finished it yet.

A friend had this display and when he switched from Marlin to Klipper he sold the screen :-( . Touch control is great, but if part of the settings there doesn't work for you, then it loses its meaning.

https://github.com/bigtreetech/BIGTREETECH-TouchScreenFirmware/issues/290
https://github.com/bigtreetech/BIGTREETECH-TouchScreenFirmware/issues/2739
https://github.com/bigtreetech/BIGTREETECH-TouchScreenFirmware/issues/1905
https://github.com/bigtreetech/BIGTREETECH-TouchScreenFirmware/issues/1555

In the past, Klipper did not support serial communication, but hopefully the new version can :-)

**Comment by vin024github** [comment]
Thank you very much Radek. Your help is greatly appreciated.I finally wakeup and found the config.ini file! It was the config file for the display not klipper config. I will check the baud rate and see if it match the 250000 baud rate set in the touchscreen.Best regardsVincent BrouilletLe 15 déc. 2023 à 01:43, Radek ***@***.***> a écrit :﻿
Sorry for my english, I'm using a translator.
BTT makes good HW but when it comes to tech support and SW it's bad.
SW for TFT is developed by a community of people (programmers) who like this display and want to improve it and publish modifications.
BBT mostly just merges changes, upgrades and fixes.
It hasn't even done that for the last year, that's why there are so many useful fixes waiting there.
Most programmers who work on the display use Marlin, so all changes are primarily compatible with Marlin.
Occasionally someone with a different system (Klipper, Smoothieware, RepRap) would sign up and want to add support for their system, but after two posts they wouldn't hear back and they hadn't finished the implementation.
The display will generally work with the Klipper. It will print and display temperature and such. but it may have a problem with some functions and settings (calibration, leveling, fan...).
Both Marlin and Klipper use the same G-codes, but some commands and some command responses have different syntax. If Klipper does not receive a command in the expected form, the display does not receive a response in the form it expects, this function will not work.
It could be solved, but it means modifying the program and compiling your own FW. But for that you need at least a basic knowledge of programming.
In the past, there was also a problem that Klipper did not display the native screen properly through the EXP connector, and BTT was working on making it operational, but I don't know if he finished it yet.
A friend had this display and when he switched from Marlin to Klipper he sold the screen :-( . Touch control is great, but if part of the settings there doesn't work for you, then it loses its meaning.
#290
#2739
#1905
#1555
